pkg/agent/api/delegatedidentity/v1/service.go

```diff
@@ -111,6 +113,63 @@ func (s *Service) isCallerAuthorized(ctx context.Context, log logrus.FieldLogger
 	return nil, status.Error(codes.PermissionDenied, "caller not configured as an authorized delegate")
 }
 
+// Attempt to attest and authorize the delegate, and then take the PID the delegate gave us
+// and attempt to attest that into a set of selectors + a subscription to changes of those selectors.
+//
+// Note that the trusted delegate is responsible for ensuring the PID is valid and not recycled,
+// from initiation of this call until the termination of the response stream, and if it is,
+// must discard any stream contents provided by this call as invalid.
+func (s *Service) SubscribeToX509SVIDsByPID(req *delegatedidentityv1.SubscribeToX509SVIDsByPIDRequest, stream delegatedidentityv1.DelegatedIdentity_SubscribeToX509SVIDsByPIDServer) error {
+	latency := adminapi.StartFirstX509SVIDUpdateLatency(s.metrics)
+	ctx := stream.Context()
+	log := rpccontext.Logger(ctx)
+	var receivedFirstUpdate bool
+
+	cachedSelectors, err := s.isCallerAuthorized(ctx, log, nil)
+	if err != nil {
+		return err
+	}
+
+	// Delegate authorized, use PID the delegate gave us to try and attest OBO
+	selectors, err := s.delegateWorkloadAttestor.Attest(ctx, int(req.Pid))
```

Comment:
Note that `Attest` takes `int` here - AFAIK PIDs are never negative, so is there a reason we didn't use `uint` here

Reply:
There are many arguments against the use of unsigned integers. All of that aside, `int` matches the type we receive when querying the linux kernel for the caller info, e.g. pid_t type on the pid field in the ucred struct. The Go stdlib also uses int's to represent process IDs. We should continue using signed integers.